src/conversion/buildviewconverter.js

```diff
@@ -291,31 +292,62 @@ class ViewConverterBuilder {
 					}
 
-					// Try to consume appropriate values from consumable values list.
-					if ( !consumable.consume( data.input, from.consume || match.match ) ) {
+					// Find allowed parent for element that we are going to insert.
+					// If current parent does not allow to insert element but one of the ancestors does
+					// then split nodes to allowed parent.
+					const splitResult = conversionApi.splitToAllowedParent( modelElement, data.position, conversionApi );
+
+					// When there is no split result it means that we can't insert element to model tree, so let's skip it.
+					if ( !splitResult ) {
 						continue;
 					}
 
-					// If everything is fine, we are ready to start the conversion.
-					// Add newly created `modelElement` to the parents stack.
-					data.context.push( modelElement );
-
-					// Convert children of converted view element and append them to `modelElement`.
-					const modelChildren = conversionApi.convertChildren( data.input, consumable, data );
-
-					for ( const child of Array.from( modelChildren ) ) {
-						writer.append( child, modelElement );
+					// Insert element on allowed position.
+					conversionApi.writer.insert( modelElement, splitResult.position );
+
+					// Convert children and insert to element.
+					const childrenOutput = conversionApi.convertChildren( data.input, consumable, Position.createAt( modelElement ) );
+
+					// Consume appropriate value from consumable values list.
+					consumable.consume( data.input, from.consume || match.match );
+
+					// Prepare conversion output range, we know that range will start before inserted element.
+					const output = new Range( Position.createBefore( modelElement ) );
+
+					// Now we need to check where the output should end.
+
+					// If we had to split parent to insert our element the we want to continue conversion
+					// inside split parent.
+					//
+					// before: <allowed><notAllowed>[]</notAllowed></allowed>
+					// after:  <allowed><notAllowed></notAllowed>[<converted></converted><notAllowed>]</notAllowed></allowed>
+					if ( splitResult.endElement ) {
+						output.end = Position.createAt( splitResult.endElement );
+
+					// When element is converted on target position (without splitting) we need to move range
+					// after this element but we need to take into consideration that children could split our
+					// element, so we need to move range after parent of the last converted child.
+					//
+					// after: <allowed>[]</allowed>
+					// before: <allowed>[<converted><child></child></converted><child></child><converted>]</converted></allowed>
+					} else if ( childrenOutput.end.parent !== childrenOutput.end.root ) {
+						output.end = Position.createAfter( childrenOutput.end.parent );
+
+					// Finally when we are sure that element and its parent was not split we need to put selection
+					// after element.
+					//
+					// after: <allowed>[]</allowed>
+					// before: <allowed>[<converted></converted>]</allowed>
```

Comment:
As above.

Reply:
Also, as far as I understand, this is a case when there are no children. Otherwise, the previous one will do the job.

Reply:
Hm... I change it yesterday.